tensorflow/compiler/mlir/tensorflow/translate/export_graphdef.cc

```diff
@@ -85,7 +85,7 @@ std::string GetName(Operation* inst) {
 
   // If the location is none of the expected types, then simply use name
   // generated using the op type.
-  return inst->getName().getStringRef().str();
+  return inst->getAttrOfType<mlir::StringAttr>("name").getValue().str();
```

Comment:
Can you add a test for this in mlir2graphdef?

Reply:
@joker-eph I have added `tensorflow/compiler/mlir/tensorflow/tests/mlir2graphdef/tf_node_name.mlir` in https://github.com/tensorflow/tensorflow/pull/30792/commits/81733c36ca0f19465886d467cb3f9a9ff8e026e9
Is it fulfill your requests?
In fact `bazel test` for mlir2graphdef doesn't pass on this branch because of the tf-mlir-translate build failure. I checked the `baze test mlir2graphde:all` passed on the commit hash:77095978f4 (where tf-mlir-translate can be build) with patches from this PR.

Reply:
Thanks! That's the kind of test I'm looking for.

Can you put only the CHECK line for the op name? 

Also the test is passing without the patch, can you double check this?

Reply:
Thank you for your comment. I pushed https://github.com/tensorflow/tensorflow/pull/30792/commits/6bde1adad2b7806b267bc8f09dc6f183e6be430c to remove extra CHECK-NEXT lines.

In my environment the mlir2graphdef tests isn't passing on this branch.
Are there anything wrong on my side?

```
$ bazel test //tensorflow/compiler/mlir/tensorflow/tests/mlir2graphdef:all
INFO: Options provided by the client:
  Inherited 'common' options: --isatty=1 --terminal_columns=181
INFO: Reading rc options for 'test' from /Users/chikanaga/work/tensorflow/.bazelrc:
  Inherited 'build' options: --apple_platform_type=macos --define framework_shared_object=true --define=use_fast_cpp_protos=true --define=allow_oversize_protos=true -c opt --announce_rc --define=grpc_no_ares=true --define=PREFIX=/usr --define=LIBDIR=$(PREFIX)/lib --define=INCLUDEDIR=$(PREFIX)/include
INFO: Reading rc options for 'test' from /Users/chikanaga/work/tensorflow/.tf_configure.bazelrc:
  Inherited 'build' options: --define with_gcp_support=true --action_env TF_NEED_CUDA=0
INFO: Analyzed 2 targets (1 packages loaded, 182 targets configured).
INFO: Found 1 target and 1 test target...
INFO: From Compiling tensorflow/core/kernels/lookup_util.cc:
tensorflow/core/kernels/lookup_util.cc:132:26: warning: 'Split' is deprecated: Use absl::StrSplit instead. [-Wdeprecated-declarations]
      tokens = str_util::Split(line, delimiter_);
                         ^
(... snip ... there are many deprecated warnings)
ERROR: /Users/chikanaga/work/tensorflow/tensorflow/compiler/mlir/tensorflow/BUILD:140:1: C++ compilation of rule '//tensorflow/compiler/mlir/tensorflow:tensorflow_dialect_registration' failed (Exit 1) wrapped_clang failed: error executing command external/local_config_cc/wrapped_clang '-D_FORTIFY_SOURCE=1' -fstack-protector -fcolor-diagnostics -Wall -Wthread-safety -Wself-assign -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG ... (remaining 124 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
In file included from tensorflow/compiler/mlir/tensorflow/ir/dialect_registration.cc:18:
./tensorflow/compiler/mlir/tensorflow/ir/tf_ops.h:126:26: error: use of undeclared identifier 'SymbolRefAttr'
    return getAttrOfType<SymbolRefAttr>("then_branch").getValue();
                         ^
./tensorflow/compiler/mlir/tensorflow/ir/tf_ops.h:130:26: error: use of undeclared identifier 'SymbolRefAttr'
    return getAttrOfType<SymbolRefAttr>("else_branch").getValue();
                         ^
./tensorflow/compiler/mlir/tensorflow/ir/tf_ops.h:165:26: error: use of undeclared identifier 'SymbolRefAttr'
    return getAttrOfType<SymbolRefAttr>("cond").getValue();
                         ^
./tensorflow/compiler/mlir/tensorflow/ir/tf_ops.h:168:26: error: use of undeclared identifier 'SymbolRefAttr'
    return getAttrOfType<SymbolRefAttr>("body").getValue();
                         ^
4 errors generated.
INFO: Elapsed time: 636.118s, Critical Path: 86.88s
INFO: 305 processes: 304 darwin-sandbox, 1 local.
FAILED: Build did NOT complete successfully

FAILED: Build did NOT complete successfully
```

Reply:
> Are there anything wrong on my side?

TF and MLIR are out-of-sync on GitHub, we're still upgrading our internal tooling to handle this automatically, sorry about this!

Reply:
BTW the workaround that was added at the top of the function probably getting in the way of verifying this patches behavior. But it would also unblock you from using the version at head (modulo the version skew that we should fix shortly in a better way). And I've filed a bug to address these.